components/raftstore-v2/src/operation/command/write/mod.rs

```diff
@@ -228,4 +233,35 @@ impl<EK: KvEngine, R> Apply<EK, R> {
         // TODO: reuse the same delete as split/merge.
         Ok(())
     }
+
+    #[inline]
+    pub fn apply_ingest(&mut self, ssts: Vec<SstMeta>) -> Result<()> {
+        PEER_WRITE_CMD_COUNTER.ingest_sst.inc();
+        let mut infos = Vec::with_capacity(ssts.len());
+        for sst in &ssts {
+            if let Err(e) = check_sst_for_ingestion(sst, self.region()) {
+                error!(
+                    self.logger,
+                    "ingest fail";
+                    "sst" => ?sst,
+                    "region" => ?self.region(),
+                    "error" => ?e
+                );
+                let _ = self.sst_importer().delete(sst);
```

Comment:
What about other ssts? They are leaked?

Reply:
What other ssts? If you referring those SSTs that pass the check, then they should be maintained by SstImporter at L262.

Reply:
What do you mean? At L251 this function returns, who will delete the other ssts that passed the check? And if someone else outside this function is deleting them, why delete one here.

Reply:
They will be cleaned up in the end by CleanupImportSst tick, which is not implemented yet.

Reply:
@lance6716 perhaps can explain the processing difference.

Reply:
Im all good as long as nothing gets leaked.

Reply:
Don't worry, this line just copies the old logic.